sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/datasources/subscriptions/SubscriptionsDbConnector.groovy

```diff
@@ -116,4 +120,32 @@ class SubscriptionsDbConnector implements SubscriptionDataSource {
           }
           return personId
     }
+
+    private boolean projectAlreadySubscribed(int subscriberId, String projectCode) {
+        String query = "SELECT id FROM subscription WHERE project_code = ? AND subscriber_id = ? "
+        Connection connection = connectionProvider.connect()
+        boolean isAlreadySubscribed = false
+
+        try{
+            connection.withCloseable {
+                def statement = connection.prepareStatement(query)
+                statement.setString(1, projectCode)
+                statement.setString(2, subscriberId.toString())
+                ResultSet resultSet = statement.executeQuery()
+                int numberOfRows = 0
+                while(resultSet.next()) {
+                    numberOfRows++
+                }
+                if (numberOfRows > 0) {
+                    isAlreadySubscribed = true
+                }
```

Comment:
```suggestion
      if(resultSet.next()) isAlreadySubscribed = true
```
You can just simply check if the query was successful and has values
https://stackoverflow.com/questions/30860186/check-if-value-exists-in-mysql-db-in-java